### PR TITLE
Feature/nl remove functionality

### LINKED ISSUE #30: Make Remove button on Reading List page remove the book from the list.

### PR DESCRIPTION
**Pull request by NicoleLatifi**
#### What's this PR do?
-  Make Remove button on Reading List page remove the book from the reading list.

Type of change made:

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Styling- no new features

#### Where should the reviewer start?

 src/Components/ReadingListBook/ReadingListBook.js 
 src/actions/index.js 
 src/reducers/ReadingList.js

#### Why is this change required? What problem does it solve?
- Allows user to remove a book directly from their Reading List page.

#### Were there any challenges that arose while implementing this feature? If so, how were they addressed?
- Yes! It was our first time removing something from store so we worked through it together as a group.

#### Any background context you want to provide?
- n/a

#### What are the relevant tickets?
- resolves #30 

#### Questions:
- n/a

### CODE REVIEW

File: src/actions/index.js [hunk 1]
```diff
@@ -3,7 +3,12 @@ export const setBooks = (books) => ({
...
 
-export const addFavorite = (readingList) => ({
-  type: 'SET_READING_LIST',
-  readingList
+export const addFavorite = (book) => ({
+  type: 'ADD_TO_READING_LIST',
+  book
+})
+
+export const removeFavorite = (id) => ({
```
Comment (kip-west) [inline src/actions/index.js:11]: So glad we figured out how to use this action object skillfully!

File: src/Components/ReadingListBook/ReadingListBook.js [hunk 2]
```diff
@@ -1,14 +1,21 @@
...
+import { bindActionCreators } from 'redux';
 
-const ReadingListBook = ({ title, image, id, removeBook }) => {
+const ReadingListBook = ({ title, image, id, removeFavorite }) => {
 
   return (
     <article className='book-card'>
       <img className="card-image" alt={title} src={image} />
-      <button className='remove-book' id={id} onClick={removeBook}>Remove</button>
+      <button className='remove-book' onClick={() => removeFavorite(id)}>Remove</button>
```
Comment (kip-west) [inline src/Components/ReadingListBook/ReadingListBook.js:12]: Watching us whittle this method down as we refactored was SO cool!!

**Review by kip-west — APPROVED** [review]

**Review by amykarnaze — APPROVED** [review]
Great job getting redux hooked up to adding and removing bookings from reading list and refactoring along the way.

File: src/reducers/ReadingList.js [hunk 3]
```diff
@@ -1,7 +1,12 @@
 export const readingList = (state = [], action) => {
   switch (action.type) {
-    case 'SET_READING_LIST':
-      return ( [...state, action.readingList] )
+    case 'ADD_TO_READING_LIST':
+      return ( [...state, action.book] )
+    case 'REMOVE_FROM_READING_LIST':
+      const newReadingList = state.filter(book => {
+        return action.id !==book.primary_isbn10
+      })
```
Comment (NakiNorton) [inline src/reducers/ReadingList.js:8]: Nice work with this! It was awesome seeing this delete functionality in action.

**Review by NakiNorton — APPROVED** [review]
So glad that worked out how to get the removeFavorite action connected and working. It was interesting to see the differences between in working with functional vs class components.

**Pull request merged**